source/common/network/io_uring_socket_handle_impl.h

```diff
@@ -0,0 +1,125 @@
+  Api::IoCallUint64Result recv(void* buffer, size_t length, int flags) override;
+  bool supportsMmsg() const override;
+  bool supportsUdpGro() const override;
+  Api::SysCallIntResult bind(Address::InstanceConstSharedPtr address) override;
+  Api::SysCallIntResult listen(int backlog) override;
+  IoHandlePtr accept(struct sockaddr* addr, socklen_t* addrlen) override;
+  Api::SysCallIntResult connect(Address::InstanceConstSharedPtr address) override;
+  Api::SysCallIntResult setOption(int level, int optname, const void* optval,
+                                  socklen_t optlen) override;
+  Api::SysCallIntResult getOption(int level, int optname, void* optval, socklen_t* optlen) override;
+  Api::SysCallIntResult ioctl(unsigned long, void*, unsigned long, void*, unsigned long,
+                              unsigned long*) override;
+  Api::SysCallIntResult setBlocking(bool blocking) override;
+  absl::optional<int> domain() override;
+  Address::InstanceConstSharedPtr localAddress() override;
+  Address::InstanceConstSharedPtr peerAddress() override;
+  void initializeFileEvent(Event::Dispatcher& dispatcher, Event::FileReadyCb cb,
+                           Event::FileTriggerType trigger, uint32_t events) override;
+  IoHandlePtr duplicate() override;
+  void activateFileEvents(uint32_t events) override;
+  void enableFileEvents(uint32_t events) override;
+  void resetFileEvents() override;
+  Api::SysCallIntResult shutdown(int how) override;
+  absl::optional<std::chrono::milliseconds> lastRoundTripTime() override { return absl::nullopt; }
+  absl::optional<uint64_t> congestionWindowInBytes() const override { return absl::nullopt; }
+  absl::optional<std::string> interfaceName() override;
+
+  // IoUringHandler
+  void onAcceptSocket(Io::AcceptedSocketParam& param) override;
+  void onRead(Io::ReadParam& param) override;
+  void onWrite(Io::WriteParam& param) override;
+
+private:
+  std::string ioUringSocketTypeStr() {
+    switch (io_uring_socket_type_) {
+    case IoUringSocketType::Unknown:
+      return "Unknown";
+    case IoUringSocketType::Client:
+      return "Client";
+    case IoUringSocketType::Server:
+      return "Server";
+    case IoUringSocketType::Listen:
+      return "Listen";
+    }
+  }
+
+  const uint32_t read_buffer_size_;
+  Io::IoUringFactory& io_uring_factory_;
+  OptRef<Io::IoUringWorker> io_uring_worker_{absl::nullopt};
+  os_fd_t fd_;
+  int socket_v6only_;
+  const absl::optional<int> domain_;
+  OptRef<Io::IoUringSocket> io_uring_socket_{absl::nullopt};
+  bool disabled_{false};
+
+  Event::FileReadyCb cb_;
+
+  OptRef<Io::AcceptedSocketParam> accepted_socket_param_{absl::nullopt};
+  OptRef<Io::ReadParam> read_param_{absl::nullopt};
+  OptRef<Io::WriteParam> write_param_{absl::nullopt};
+  IoUringSocketType io_uring_socket_type_{IoUringSocketType::Unknown};
+
+  std::unique_ptr<IoHandle> shadow_io_handle_;
+  bool enable_server_socket_{true};
+  bool enable_client_socket_{true};
```

Comment:
note for the reviewer: the shade iohandle is just for debugging, it is enabled to debug accept/server/client socket separately. Like you can disable server/client socket to use normal iohandle, then debug io_uring accept socket.